File: ContactManagerDIExercise/app/src/main/java/com/anushka/androidtutz/contactmanager/di/component/AppComponent.java
package com.anushka.androidtutz.contactmanager.di.component;

import com.anushka.androidtutz.contactmanager.MainActivity;
import com.anushka.androidtutz.contactmanager.di.module.AppModule;
import com.anushka.androidtutz.contactmanager.di.module.RepositoryModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, RepositoryModule.class})
public interface AppComponent {

    void inject(MainActivity activity);
}

File: TMDBClient2/app/src/main/java/com/example/tmdbclient/presentation/di/core/AppComponent.kt
package com.example.tmdbclient.presentation.di.core

import com.example.tmdbclient.presentation.di.artist.ArtistSubComponent
import com.example.tmdbclient.presentation.di.movie.MovieSubComponent
import com.example.tmdbclient.presentation.di.tvshow.TvShowSubComponent
import dagger.Component
import javax.inject.Singleton

@Component(
    modules = [
        AppModule::class,
        CacheDataModule::class,
        DataBaseModule::class,
        LocalDataModule::class,
        NetModule::class,
        RemoteDataModule::class,
        RepositoryModule::class,
        UseCaseModule::class
    ]
)
@Singleton
interface AppComponent {

    fun movieSubComponent(): MovieSubComponent.Factory
    fun tvShowSubComponent(): TvShowSubComponent.Factory
    fun artistSubComponent(): ArtistSubComponent.Factory
}